### PR TITLE
Poisons

### PR DESCRIPTION
**Pull request by JFranklin3000**
##### In this PR
- Added Poisons options to weapons
- Created all the Poisons in the Cobra Codex
- created a way to do training and qualifications

##### Testing
- Added list below to increase poison and toxin training and qualifications. 
- validate poisons in the Cobra Codex
image

### CODE REVIEW

File: templates/actor/parts/misc/training.hbs [hunk 1]
```diff
@@ -15,6 +15,16 @@
           {{ localize "E20.ItemTypeWeaponPlural" }}
         </th>
         {{/ifNotEquals}}
+        {{#if @root.actor.system.poisonTraining}}
+          <th class="tg-td">
+            {{ localize "E20.ItemTypePoisonPlural" }}
+          </th>
+          {{#ifCond @root.actor.system.poisonTraining '>' 4}}
```
Comment (phildominguez) [inline templates/actor/parts/misc/training.hbs:22]: I'm not big on having such specific logic in the hbs... could we instead generate these strings in `_preparePoisonTraining`?

File: templates/actor/parts/misc/training.hbs [hunk 2]
```diff
@@ -23,27 +33,50 @@
...
           {{/if}}
         </td>
         {{#ifNotEquals @root.role.system.version 'powerRangers'}}
         <td class="tg-td">
-          {{listProficiencies @root.system.trained.weapons @root.config.weaponTypes}}
+          {{formatBooleanList @root.system.trained.weapons @root.config.weaponTypes 'unit'}}
         </td>
         {{/ifNotEquals}}
+        {{#if @root.actor.system.poisonTraining}}
+          <td>
```
Comment (phildominguez) [inline templates/actor/parts/misc/training.hbs:47]:
```suggestion
          <td class="tg-td">
```
And below

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: templates/actor/parts/misc/training.hbs (hunk 1 above)
Comment (JFranklin3000) [reply templates/actor/parts/misc/training.hbs:22]: It is a weird one, but I can do a ``{{#if @root.actor.system.trained.toxins.all}}`` instead actually.

File: lang/en.json [hunk 3]
```diff
@@ -725,7 +738,10 @@
     "WeaponAvailability": "Availability",
     "WeaponEffect": "Effect",
     "WeaponHands": "Hands",
+    "WeaponIsPoison": "Is a poison?",
     "WeaponUsesPerScene": "Uses per scene",
+    "WeaponPoisonSubType": "Subtype",
+    "WeaponPoisonType": "Type",
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply lang/en.json:744]: No issue with that changed.

File: lang/en.json [hunk 4]
```diff
@@ -725,7 +738,10 @@
     "WeaponAvailability": "Availability",
     "WeaponEffect": "Effect",
     "WeaponHands": "Hands",
+    "WeaponIsPoison": "Is a poison?",
     "WeaponUsesPerScene": "Uses per scene",
+    "WeaponPoisonSubType": "Subtype",
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply lang/en.json:743]: They call it how you apply it in the book, so Poison Application?

File: templates/actor/parts/misc/training.hbs [hunk 5]
```diff
@@ -23,27 +33,50 @@
...
-          {{listProficiencies @root.system.qualified.armors @root.config.armorTypes}}
+          {{formatBooleanList @root.system.qualified.armors @root.config.armorTypes 'unit'}}
         </td>
         <td class="tg-td">
-          {{listProficiencies @root.system.qualified.weapons @root.config.weaponTypes}}
+          {{formatBooleanList @root.system.qualified.weapons @root.config.weaponTypes 'unit'}}
         </td>
-      </tr>
+        {{#if @root.actor.system.poisonTraining}}
+          <td>
```
Comment (phildominguez) [inline templates/actor/parts/misc/training.hbs:71]:
```suggestion
          <td class="tg-td">
```
These are what makes them centered

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: templates/actor/parts/misc/training.hbs [hunk 6]
```diff
@@ -23,27 +33,50 @@
...
+            {{else}}
+              {{formatBooleanList @root.system.trained.poisons @root.config.poisonTraining 'unit'}}
+            {{/if}}
+          </td>
+          {{#if @root.actor.system.trained.toxins.all}}
+            <td>
+              {{localize "E20.PoisonTrainingAll"}}
+            </td>
+          {{/if}}
+        {{/if}}
```
Comment (phildominguez) [inline templates/actor/parts/misc/training.hbs:59]:
```suggestion
        <td class="tg-td">
        {{#if @root.actor.system.poisonTraining}}
            {{#if @root.actor.system.trained.poisons.all}}
              {{localize "E20.PoisonTrainingAll"}}
            {{else}}
              {{formatBooleanList @root.system.trained.poisons @root.config.poisonTraining 'unit'}}
            {{/if}}
          {{#if @root.actor.system.trained.toxins.all}}
              {{localize "E20.PoisonTrainingAll"}}
          {{/if}}
        {{/if}}
        </td>
```

**Review by phildominguez — CHANGES_REQUESTED** [review]

**Review by phildominguez — APPROVED** [review]

**Pull request merged**